### ISSUE TITLE
Documentation is Horrible

### ISSUE THREAD

**Issue by graftedin** [p0lym0rphik/cakephp-metabehavior#5]
What does MyModel mean?  Is this the name of the meta model so by default that would be MetaData?  Or is MyModel the base model so if I'm saving a post I'm going to use the Post model and then some how metabehavior is going to magically figure out the extra data and submit that to MetaData?  If its the 2nd option this plugin does not work.  And honestly if this thing can't sniff out the extra data like I'm suggesting in the 2nd option then its not very useful.

**Comment by p0lym0rphik** [comment]
MyModel is the base model where metadas has to be applyed. As you can see in documentation, you have to declare the meta behavior in model. This plugin work on projects, so if you want i help you, you have to show a part of your code...

**Comment by graftedin** [comment]
I have a model called Users.  By your description I think MyModel should be the same as my Users model.  The Users database table has these fields: id, name, email.  However, if my form has a field for phone I'd expect that as I submit it the model should use the meta behavior and store the phone number in MetaData (meta_data).  That is not happening.

I used your code for an other project about a year ago and I had to make major changes to make it work.  However, I was new to cake at the time so I'm trying to modify as little as I have to this time.

What I have done is put print statements in the behaviors afterSave because that seems to be how it hooks in to my model.  My statements are printing however at "Ensure there is data." its not finding data because its looking for data in the MetaData model rather than the Users model.

The only think I can think of is I need to manually move the extra data to the MetaData model so that afterSave will find it.  Maybe I need to change the fields in my form so it submits that data to MetaData instead?  If that's the case your documentation doesn't say that.

My previous project I had a Meta model for every table I wanted to allow meta on and I linked them together with hasMany.  So for my example I had UserMeta as an actual model and linked Users and UserMeta with hasMany.  However your documentation doesn't say I need to use hasMany.  Your documentation also doesn't say I should have a meta model for every table I need the meta behavior on.

The bottom line.  Your documentation along with most of cake's documentation never fully explains the solutions.  The explanation is very brief and often assumes that the user knows more than they do.  If you take the time to show us the database tables and fields, the fields submitted and everything needed to make your behavior work then idiots like me won't need to bug you because we can't get your code to work.
